codechain/rpc.rs

```diff
@@ -32,39 +32,27 @@ pub struct RpcHttpConfig {
     pub hosts: Option<Vec<String>>,
 }
 
-pub fn rpc_http_start(
-    cfg: RpcHttpConfig,
-    enable_devel_api: bool,
-    deps: &rpc_apis::ApiDependencies,
-) -> Result<HttpServer, String> {
-    let url = format!("{}:{}", cfg.interface, cfg.port);
+pub fn rpc_http_start(config: &Config, deps: &rpc_apis::ApiDependencies) -> Result<HttpServer, String> {
```

Comment:
I have a question about this refactoring.
Why do you think using `Config` is better than using the specific configuration trait?

Reply:
`ApiDependencies::extend_api` receives `Config` as an argument, so we need to pass it anyways.
Speciailized configs (e.g. `RpcHttpConfig`) can be derived from `Config`, so I thought it'd be better to reduce duplication.

Reply:
I think it would be better to refactor the code to be:
```rust
let rpc_server = {
  if !config.rpc.disable.unwrap() {
    let rpc_server = setup_rpc_server(config, deps);
    Some(rpc_http_start(rpc_server, config.rpc_http_config()))
  } else {
    None
  }
};
```

Reply:
It becomes a bit verbose but I like the idea here. @sgkim126 What do you think about it?

Reply:
I like @foriequal0's idea too.

Reply:
Applied the suggestion.